R/rintrinio.R

```diff
@@ -1,29 +1,140 @@
+#' Given the ticker, statement, year(s) and period(s) return the complete available financial information from the Intrinio API stock data 
 #'
 #' @param api_key character (sandbox or production) from Intrinio
-#' @param ticker ticker symbol
+#' @param ticker character ticker symbol
 #' @param statement character that represents the financial statement that you want to study
-#' @param year vector of the years (as characters) you want the information for
-#' @param period vector of the periods (characters) you want the information from
+#' @param year character vector of the year(s) you want the information for
+#' @param period character vector of the period(s) you want the information for
 #'
-#' @return a dataframe that contains information about the given statement for a given ticker for the specified years
+#' @return a dataframe that contains financial statement about the company(ticker) for the specified time frame
 
 #' @export
 #' @examples
-#' gather_financial_statement_time_series(api_key, 'AAPL', 'income_statement', ['2018,'2019'], ['Q1'])
+#' gather_financial_statement_time_series(api_key, 'CVX', 'income_statement', c('2017,'2018'), c('Q1','Q3'))
 
 gather_financial_statement_time_series <- function(api_key, ticker, statement, year, period){
-  tibble()
+  
+  ## Limited free access on intrino provides only Sandbox access
+  
+  ## Checks
+  
+  if (typeof(api_key) != "character"){
+    stop("The api_key must be a string")
+  }
+  
+  if (typeof(ticker) != "character") {
+    stop("The ticker must be a string. For ex. 'CSCO'")
+  }
+  
+  if (typeof(statement) != "character") {
+    stop("The statement must be a string. For ex. 'cash_flow_statement'")
+  }
+  
+  if (typeof(year) != "character") {
+    stop("The year must be a character vector. For ex. c('2016', '2018')")
+  }
+  
+  for(y in year){
+    if(nchar(y) != 4){
+      stop("Sorry, year must be a string of 4 digits")
+    }
+  }
+  
+  if (typeof(period) != "character") {
+    stop("The year must be a character vector. For ex. c('Q1', 'Q3')")
+  }
+  
+  available_statements <- c('income_statement', 'cash_flow_statement', 'balance_sheet_statement')
+  
+  `%notin%` <- Negate(`%in%`)
+  
+  if (statement %notin% c('income_statement', 'cash_flow_statement', 'balance_sheet_statement')) {
+    stop("Valid entries for statement are 'income_statement', 'cash_flow_statement' or 'balance_sheet_statement'")
+  }
+  
+  ## List of available tickers with sandbox key: 
+  ## https://product.intrinio.com/developer-sandbox/coverage/us-fundamentals-financials-metrics-ratios-stock-prices
+  
+  available_tickers <- c('AAPL', 'AXP', 'BA', 'CAT', 'CSCO', 'CVX', 'DIS', 'DWDP', 'GE', 'GS', 'HD', 'IBM', 'INTC', 'JNJ', 'JPM', 'KO', 'MCD', 'MMM', 'MRK', 'MSFT', 'NKE', 'PFE', 'PG', 'TRV', 'UNH', 'UTX', 'V', 'VZ', 'WMT', 'XOM')
+  
+  if (ticker %notin% available_tickers) {
+    stop("Valid entries for ticker provided in the Readme.md")
+  }
+  
```

Comment:
Sam told me that the people who pay actually have more tickers available, so we shouldn't limit the possibles tickers

```suggestion  
```